modules/components/src/AdvancedFacetView/index.js

```diff
@@ -47,178 +129,188 @@ export default class AdvancedFacetView extends React.Component {
+                  {/* using a thin local state here for rendering performance optimization */}
+                  <Component initialState={{ value: searchTerm || '' }}>
+                    {({ state: { value }, setState }) => (
+                      <TextInput
+                        icon={<FaFilter />}
+                        rightIcon={
+                          <FaTimesCircleO
+                            onClick={() => {
+                              setState({ value: null }, () => {
+                                this.setState({
+                                  searchTerm: null,
+                                });
+                              });
+                            }}
+                          />
+                        }
+                        className="filterInput"
+                        type="text"
+                        placeholder="Filter"
+                        value={value || ''}
+                        showClearButton={true}
+                        onChange={e => {
+                          const value = e.target.value;
+                          setState({ value }, () => {
+                            this.setSearchTerm(value);
+                          });
+                        }}
+                      />
+                    )}
+                  </Component>
+                  <div
+                    className={css`
+                      display: flex;
+                      flex: 1;
+                      height: 100%;
+                    `}
+                  >
+                    {statComponent}
+                  </div>
+                </div>
+                <div className={`facets`}>
+                  <FacetView
+                    extendedMapping={extendedMapping}
+                    valueCharacterLimit={valueCharacterLimit}
+                    constructEntryId={this.constructFilterId}
+                    ref={view => (this.facetView = view)}
+                    sqon={sqon}
+                    onValueChange={this.handleSqonChange}
+                    aggregations={aggregations}
+                    searchString={searchTerm}
+                    displayTreeData={filterDisplayTreeDataBySearchTerm({
+                      displayTree: visibleDisplayTreeData,
                       aggregations,
-                    }).displayTreeDataWithValue
-                  : displayTreeData
-              }
-            />
+                      searchTerm: searchTerm,
+                    })}
+                  />
+                </div>
+              </div>
+            </div>
+          </>
+        )}
+        {isLoading && (
+          <div
+            className={`loadingScreen ${css`
+              position: absolute;
+              left: 0px;
+              right: 0px;
+              top: 0px;
+              bottom: 0px;
+              display: flex;
+              justify-content: center;
+              align-items: center;
+            `}`}
+          >
```

Comment:
not sure, but would the loading screen be worth componentizing for re-use as well?